app/controllers/pages_controller.rb

```diff
@@ -142,4 +131,23 @@ def search_params
     params[:search] ||= {}
     params[:search].reverse_merge default_params
   end
+
+  def process_one_click
+    @process_one_click ||= PaymentProcessor::Braintree::OneClickFromUri.new(
+      params,
+      page: @page,
+      member: recognized_member,
+      cookied_payment_methods: cookies.signed[:payment_methods]
+    ).process
+  end
+
+  def redirect_unless_published
+    redirect_to(Settings.home_page_url) unless @page.published? || user_signed_in?
+  end
+
+  def localize
+    if @page && @page.language.present?
+      set_locale(@page.language.code)
```

Comment:
The `set_locale` that was removed was in LiquidRenderer, and that version was not used from anywhere. The `set_locale` that's used here is implemented in `ApplicationController`